src/transformation/visitors/language-extensions/table.ts

```diff
@@ -172,3 +181,29 @@ function transformTableSetExpression(
     );
     return lua.createNilLiteral();
 }
+
+function transformTableAddExpression(
+    context: TransformationContext,
+    node: ts.CallExpression,
+    extensionKind: extensions.ExtensionKind
+): lua.Expression {
+    const args = node.arguments.slice();
+    if (
+        extensionKind === extensions.ExtensionKind.TableAddMethodType &&
+        (ts.isPropertyAccessExpression(node.expression) || ts.isElementAccessExpression(node.expression))
+    ) {
+        // In case of method (no table argument), push method owner to front of args list
+        args.unshift(node.expression.expression);
+    }
+
+    // arg0[arg1] = true
+    const [table, value] = transformExpressionList(context, args);
+    context.addPrecedingStatements(
+        lua.createAssignmentStatement(
+            lua.createTableIndexExpression(table, value),
+            lua.createBooleanLiteral(true),
+            node
+        )
+    );
+    return lua.createNilLiteral();
```

Comment:
Any thoughts on returning `table` instead? This is how sets work in TS:
```ts
const myset = new Set<number>().add(1).add(2).add(3);
```
Though I guess that could get us in an annoying spot where we have to worry about side effects evaluating `table`, so I'd be fine leaving this as it is.

Reply:
This could be possible to implement, however it would be inconsistent with existing lua table extension functions.
Perhaps this could be done in another PR.